### DISCUSSION TITLE
Task Group failure/success semantics

### DISCUSSION THREAD

**Discussion by RonaldGalea** [apache/airflow#34522] [General]
I encountered a task group behaviour that seems counter-intuitive. In the DAG below I define a simple task group that will have one failing and one successful task, as a result the task group is marked as failed. However, the last task still executed and the overall DAG is marked successful:

```
with DAG(
    "task_groups_example",
    start_date=datetime(2021, 1, 1),
    catchup=False,
    schedule_interval=None,
) as dag:

    @task_group()
    def group() -> None:
        @task()
        def fail():
            raise ValueError("This task always fails")

        @task(trigger_rule="one_failed")
        def succeed() -> None:
            print("This task always succeeds")

        fail() >> succeed()

    @task()
    def end_node() -> None:
        print("End node")

    group() >> end_node()
```

![image](https://github.com/apache/airflow/assets/41665873/01b4e924-af6d-4865-80c1-3b9440fb4e51)

If a regular task fails, downstream tasks will not be executed and the DAG will be marked as failed. Why is it different with task groups?

It seems that the dependency is only connected to the last task executing in the group - but what is the meaning of the group failure/success then?

To note, setting up the dependencies slightly differently behaves as expected:
```

@task_group()
    def group() -> None:
        @task()
        def fail():
            raise ValueError("This task always fails")

        @task()
        def succeed() -> None:
            print("This task always succeeds")

        fail()
        succeed()
```

**Comment by Taragolis** [comment]
I’m not a confident to answer for question ‘but what is the meaning of the group failure/success then?’. I could only tell about the difference between TaskGroup, TaskInstance and DagRun.
1. TI and DagRun it is objects with sets of rules, own statuses and more important the data about this object persist in DB
2. TaskGroup it is just namespace for tasks with some additional features however it doesn’t have own statuses and doesn’t have own tables, and information stored only as part of TaskInstance
